crates/rustc_codegen_spirv/src/codegen_cx/constant.rs

```diff
@@ -481,17 +482,20 @@ impl<'tcx> CodegenCx<'tcx> {
                 );
                 result
             }
-            SpirvType::Pointer { .. } => {
-                let ptr = self.read_alloc_ptr(alloc, offset);
-                self.scalar_to_backend(
-                    ptr.into(),
-                    &abi::Scalar {
-                        value: Primitive::Pointer,
-                        valid_range: 0..=!0,
-                    },
-                    ty,
-                )
-            }
+            SpirvType::Pointer { .. } => match self.read_alloc_ptr(alloc, offset) {
+                Some(ptr) => match ptr {
+                    Scalar::Ptr(ptr) => self.scalar_to_backend(
+                        ptr.into(),
+                        &abi::Scalar {
+                            value: Primitive::Pointer,
+                            valid_range: 0..=!0,
+                        },
+                        ty,
+                    ),
+                    Scalar::Int(_) => self.constant_null(ty),
+                },
+                None => self.const_undef(ty),
```

Comment:
This should probably be `zombie`'d as per https://github.com/EmbarkStudios/rust-gpu/issues/209#issuecomment-742771895 - at least I don't think a plain undef value is correct, assuming SPIR-V used any of the existing meanings of "undef".